.github/workflows/trivy.yaml

```diff
@@ -0,0 +1,49 @@
+name: Trivy
+
+on:
+  push:
+    branches:
+      - "main"
+  schedule:
+    - cron: '0 20 * * *'
+
+permissions:
+  contents: read
+
+jobs:
+  image-scan:
+    permissions:
+      contents: read  # Required for actions/checkout to fetch code
+    name: Image Scan
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+
+      - name: Build Controller Docker image
+        run: |
+          make docker-build.controller TAG=${{ github.sha }}
+
+      - name: Build Ext-Proc Docker image
+        run: |
+          make docker-build.extproc TAG=${{ github.sha }}
+
+      - name: Run Trivy vulnerability scanner for Controller
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ghcr.io/envoyproxy/ai-gateway/controller:${{ github.sha }}
+          format: 'table'
+          severity: 'CRITICAL,HIGH,MEDIUM,LOW'
+          vuln-type: 'os,library'
+          exit-code: '1'  # Fail workflow on detected vulnerabilities
+          ignore-unfixed: true  # Ignore unfixed vulnerabilities
```

Comment:
can you make "controller" and "extproc" as part of matrix so that we could remove the duplication? https://github.com/envoyproxy/ai-gateway/blob/8aa1b4b197f69aef74c8b7c4abc16f6947ee5ff3/.github/workflows/docker_builds_template.yaml#L11-L14